src/Analyser/TypeSpecifier.php

```diff
@@ -347,6 +348,59 @@ public function specifyTypesInCondition(
 					$rootExpr,
 				);
 			}
+			if (
```

Comment:
if you move this block into the upper `if ($expressions !== null) {` block, it would work independent of operand order.

atm the code only works for ` if ($other::class === static::class) {`, within the upper IF it would also work for ` if (static::class === $other::class) {`